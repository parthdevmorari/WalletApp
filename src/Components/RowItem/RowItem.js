import React from "react";
import { View, Text,Image,Switch } from "react-native";
import styles from "./RowItemStyle";

const RowItem = (props) => {

	const { subTitle,title,value,onValueChange,imageSource,isSwitch,inactiveTrackColor,activeThumbColor,inactiveThumbColor,activeTrackColor, containerStyle } = props;

		return(
			<View style={[styles.rowContainer,containerStyle ]} >
                	
                <View style={styles.row}>

                    <View style={styles.iconContainer}>
                        <Image source={imageSource} style={[styles.imgStyle]} />
                    </View>
                    <View style={styles.containeContainer}>
                        <Text style={[styles.titleTxt]}>{title}</Text>
                        <Text style={[styles.subTitleTxt]}>{subTitle}</Text>
                    </View>
                </View>

               {isSwitch && 
               <View style={styles.switchContainer}>
                    <Switch
                        trackColor={{ false: inactiveTrackColor, true: activeTrackColor }}
                        thumbColor={value ?inactiveThumbColor : activeThumbColor}
                        onValueChange={onValueChange}
                        value={value}
                        style={[styles.swtichStyle]}
                        
                    />
                </View>}
	
			 </View>
               
		)
	
}

export default RowItem;

