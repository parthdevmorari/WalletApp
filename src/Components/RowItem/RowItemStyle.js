import { StyleSheet, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color'
import { theme } from '../../Constants/Theme'
const styles = StyleSheet.create({

	rowContainer: {
		flexDirection: "row",
		justifyContent: "space-between",
		marginHorizontal:wp(6),
		paddingVertical: wp(4),

	},
	row: {
		flexDirection: "row",
	},
	iconContainer : {
		width: 32,
		height: 32,
	},
	containeContainer : {
		paddingLeft: wp(2.5),
	},
	switchContainer : {
		justifyContent: "flex-start",
		
	},
	
	swtichStyle: {
		fontSize:wp(2),
		width: wp(8),
		height: wp(2),
		marginRight: wp(6),
	},
	titleTxt: {
		fontSize : wp(3.5),
		textAlign: "left",
		fontWeight: "bold",
		color: color._25345F,
		paddingBottom:wp(1)
	},
	subTitleTxt: {
		fontSize : wp(3),
		textAlign: "left",
		color:color._222222
	},


});
export default styles;