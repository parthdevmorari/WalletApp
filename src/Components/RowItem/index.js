import component from './RowItem';
export default component;