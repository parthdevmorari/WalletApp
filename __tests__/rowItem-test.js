import 'react-native';
import React from 'react';
import RowItem from '../src/Components/RowItem/RowItem'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('RowItem Component', () => {
  test('RowItem Component', () => {
    const wrapper = shallow(<RowItem />);
    expect(wrapper.exists()).toBe(true);
  });
  it('RowItem  toMatchSnapshot', () => {
    const wrapper = shallow(<RowItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});