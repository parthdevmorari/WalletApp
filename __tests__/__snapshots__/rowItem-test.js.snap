// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowItem Component RowItem  toMatchSnapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 45,
        "paddingVertical": 30,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    >
      <Image
        style={
          Array [
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 19,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#25345F",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "paddingBottom": 7.5,
              "textAlign": "left",
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#222222",
              "fontSize": 22.5,
              "textAlign": "left",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
